projects/components/src/open-in-new-tab/open-in-new-tab.component.ts

```diff
@@ -0,0 +1,46 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import {
+  ExternalNavigationWindowHandling,
+  NavigationParamsType,
+  NavigationService,
+  TimeRangeService
+} from '@hypertrace/common';
+import { ButtonSize, ButtonStyle } from '../button/button';
+
+@Component({
+  selector: 'ht-open-in-new-tab',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="open-in-new-tab" htTooltip="Open in a new tab">
+      <ht-button
+        class="open-in-new-tab-button"
+        display="${ButtonStyle.Outlined}"
+        [size]="this.size"
+        icon="${IconType.OpenInNewTab}"
+        (click)="this.openInNewTab()"
+      ></ht-button>
+    </div>
+  `
+})
+export class OpenInNewTabComponent {
+  @Input()
+  public size?: ButtonSize = ButtonSize.Medium;
+
+  @Input()
+  public url?: string;
+
+  public constructor(
+    private readonly navigationService: NavigationService,
+    private readonly timeRangeService: TimeRangeService
+  ) {}
+
+  public openInNewTab(): void {
+    this.navigationService.navigate({
+      navType: NavigationParamsType.External,
+      windowHandling: ExternalNavigationWindowHandling.NewWindow,
+      // Use input url if available. Else construct a shareable URL for the page
```

Comment:
Is the else part valid or should we throw an error if the url is missing?

Reply:
like, would we ever have a case when we would want to open the current url in a new tab through a button?

Reply:
I think its good to have. Example, I'm on the security detail page I want to open this in new tab and keep the history on the current tab. I think this button would be handy compared to copy-pasting the link(which isn't a permalink)  from address bar.